### PR TITLE
Feature/7580 additional services post route

### PR DESCRIPTION
**Pull request by alibombaye**
Story [AB#7850](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7850)
Task [AB#8354](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8354) [AB#8355](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8355) [AB#8334](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8334)

Screenshot 2020-07-12 at 15 03 19

### CODE REVIEW

File: app/pages/sections/additional-services/order-item/controller.js [hunk 1]
```diff
@@ -71,105 +64,30 @@ export const getOrderItemErrorPageContext = (params) => {
...
-
-  await putData({
-    endpoint, body, accessToken, logger,
-  });
-  logger.info(`Order item successfully updated for order id: ${orderId} and order item id: ${orderItemId}`);
-  return { success: true };
-};
-
-export const saveSolutionOrderItem = async ({
+export const saveOrderItem = async ({
```
Comment (sam-roberts-bjss) [inline app/pages/sections/additional-services/order-item/controller.js:67]: Wow much neater!

File: app/pages/sections/additional-services/order-item/controller.js [hunk 2]
```diff
@@ -181,7 +99,7 @@ export const saveSolutionOrderItem = async ({
       logger.info(`Follow validation errors returned from the API ${JSON.stringify(err.response.data.errors)}`);
```
Comment (sam-roberts-bjss) [inline app/pages/sections/additional-services/order-item/controller.js:99]: Not part of your change but should that say 'Follow ...'? I am thinking 'Following... ' but could be wrong.

File: app/pages/sections/additional-services/order-item/controller.js (hunk 1 above)
Comment (alibombaye) [reply app/pages/sections/additional-services/order-item/controller.js:67]: I can't wait to get my hands on the version of this in catalogue-solutions and retrofit it.

File: app/pages/sections/additional-services/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js [hunk 3]
```diff
@@ -148,106 +148,102 @@ test('should render the delete button as not disabled', async (t) => {
...
+    .expect(errorSummary.exists).ok()
+    .expect(errorSummary.find('li a').count).eql(1)
+    .expect(await extractInnerText(errorSummary.find('li a').nth(0))).eql(content.errorMessages.PriceRequired)
+    .expect(errorMessage.exists).ok()
+    .expect(await extractInnerText(errorMessage)).eql('Error:')
+
+    .expect(price.hasClass('nhsuk-input--error')).ok();
+});
+
+test('should navigate to catalogue solution dashboard page if save button is clicked and data is valid', async (t) => {
```
Comment (sam-roberts-bjss) [inline app/pages/sections/additional-services/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js:201]: Navigate to additional services dashboard?

File: app/pages/sections/additional-services/routes.js [hunk 4]
```diff
@@ -67,5 +72,54 @@ export const additionalServicesRoutes = (authProvider, addContext, sessionManage
...
+        serviceRecipientId: pageData.serviceRecipientId,
+        serviceRecipientName: pageData.serviceRecipientName,
+        itemId: pageData.itemId,
+        itemName: pageData.itemName,
+        selectedPrice: pageData.selectedPrice,
+        formData: req.body,
+      });
+
+      if (apiResponse.success) {
+        logger.info('redirecting additional-services main page');
```
Comment (sam-roberts-bjss) [inline app/pages/sections/additional-services/routes.js:103]: 'redirecting additional-services main page' -> 'redirecting to the additional-services main page' ?

File: app/pages/sections/additional-services/routes.test.js [hunk 5]
```diff
@@ -11,6 +11,7 @@ import {
 } from 'buying-catalogue-library';
```
Comment (sam-roberts-bjss) [inline app/pages/sections/additional-services/routes.test.js:11]: Couldn't see a mock reset call in this file. Is one needed?

File: app/pages/sections/additional-services/order-item/controller.js (hunk 2 above)
Comment (alibombaye) [reply app/pages/sections/additional-services/order-item/controller.js:99]: Got rid of the follow now reads as  `Validation errors ...`

File: app/pages/sections/additional-services/routes.test.js (hunk 5 above)
Comment (alibombaye) [reply app/pages/sections/additional-services/routes.test.js:11]: oooh I wonder what can of worms that will open 😂

Comment (alibombaye) [reply app/pages/sections/additional-services/routes.test.js:11]: wow no worms.. I've added a resetAllMocks to this.

Comment (sam-roberts-bjss) [reply app/pages/sections/additional-services/routes.test.js:11]: Sweet!

**Pull request merged**